src/hypofuzz/dashboard.py

```diff
@@ -292,20 +364,49 @@ def send_nowait_from_anywhere(msg: ListenerEventT) -> None:
 
     db = get_db()
     # load initial database state before starting dashboard
-    for key in db.fetch(b"hypofuzz-test-keys"):
-        reports = db.fetch_reports(key)
-        metadata = db.fetch_metadata(key)
-        if reports:
-            # TODO we should really add the node id to hypofuzz-test-keys entries
-            node_id = reports[0].nodeid
-            REPORTS[node_id] = SortedList(reports, key=lambda r: r.elapsed_time)
-        if metadata:
-            node_id = metadata[0].nodeid
-            # there is usually only a single metadata, unless we updated right
-            # as the db is inserting a new one. TODO take latest by
-            # elapsed_time, when we have Metadata extend Report
-            METADATA[node_id] = metadata[0]
+    for fuzz_target in COLLECTION_RESULT.fuzz_targets:
+        # a fuzz target (= node id) may have many database keys over time as the
+        # source code of the test changes. Show only reports from the latest
+        # database key = source code version.
+        #
+        # We may eventually want to track the database_key history of a node id
+        # and at least transfer its covering corpus across when we detect a migration,
+        # as well as possibly showing a history ui in the dashboard.
```

Comment:
Hmm, maybe `hypofuzz-test-keys` is the mechanism which would let us do this?  i.e. collect by nodeid as here, then load reports from any other test-keys, and group them in if the nodeid was from a collected test.

Happy to leave it for a follow-up though.